ansible/roles/linux-perf/tasks/main.yml

```diff
@@ -0,0 +1,64 @@
+---
+
+#
+# installs Linux perf @ /usr/local/bin/perf
+#
+
+- stat: path=/usr/local/bin/perf
+  register: linux_perf
+  tags: linux_perf
+
+- name: Clean path
+  file:
+    state: absent
+    path: /data/tmp/
+  when: linux_perf.stat.exists == False
+  tags: linux_perf
+
+- name: Download Linux source code
+  git:
+    repo: https://github.com/torvalds/linux.git
+    depth: 1
+    dest: /data/tmp/linux
+  when: linux_perf.stat.exists == False
+  tags: linux_perf
+
+- name: Install build-essential
```

Comment:
You can combine all of these `package` calls into a single one, with a loop. An example is in the `benchmarking` role's `tasks/main.yml` file